go-controller/pkg/node/gateway_udn.go

```diff
@@ -0,0 +1,154 @@
+package node
+
+import (
+	"fmt"
+	"net"
+	"strings"
+
+	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/config"
+	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/types"
+	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/util"
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/klog/v2"
+	utilnet "k8s.io/utils/net"
+)
+
+// UserDefinedNetworkGateway contains information
+// required to program a UDN at each node's
+// gateway.
+// NOTE: Currently invoked only for primary networks.
+type UserDefinedNetworkGateway struct {
+	// network information
+	util.NetInfo
+	// stores the networkID of this network
+	networkID int
+	// node that its programming things on
+	node *v1.Node
+}
+
+func NewUserDefinedNetworkGateway(netInfo util.NetInfo, networkID int, node *v1.Node) *UserDefinedNetworkGateway {
+	return &UserDefinedNetworkGateway{
+		NetInfo:   netInfo,
+		networkID: networkID,
+		node:      node,
+	}
+}
+
+// AddNetwork will be responsible to create all plumbings
+// required by this UDN on the gateway side
+func (udng *UserDefinedNetworkGateway) AddNetwork() error {
+	return udng.addUDNManagementPort()
```

Comment:
shouldn't we add mgmt port only for `config.Gateway.Mode == config.GatewayModeLocal` case ?

Reply:
think of it this way => in default network today its the same the port exists for both modes.. its going to be the same for UDNs as well